### PR TITLE
Upsert Queue: activity implementation

### PR DESCRIPTION
**Pull request by spolu**
## Description

- Tweaks the `EnqueueUpsertDocument` definition to match activity needs
- Factorize the postUpsertHook code
- Implement the activity

## Risk

N/A

## Deploy Plan

- deploy `front`

### CODE REVIEW

File: front/upsert_queue/temporal/activities.ts [hunk 1]
```diff
@@ -1,7 +1,98 @@
...
-  mainLogger.info({ upsertQueueId }, "Retrieving upsert queue item");
+  if (!DUST_UPSERT_QUEUE_BUCKET) {
+    throw new Error("DUST_UPSERT_QUEUE_BUCKET is not set");
+  }
+  if (!SERVICE_ACCOUNT) {
+    throw new Error("SERVICE_ACCOUNT is not set");
+  }
+
+  let logger = mainLogger.child({ upsertQueueId });
+  logger.info("[UpsertQueue] Retrieving item");
```
Comment (lasryaric) [inline front/upsert_queue/temporal/activities.ts:25]: isn't that missing a `{}` ?

File: front/lib/upsert_document.ts [hunk 2]
```diff
@@ -1,28 +1,44 @@
...
+const { DUST_UPSERT_QUEUE_BUCKET, SERVICE_ACCOUNT } = process.env;
+
+export const EnqueueUpsertDocument = t.type({
   workspaceId: t.string,
   dataSourceName: t.string,
   documentId: t.string,
   tags: t.union([t.array(t.string), t.null]),
   parents: t.union([t.array(t.string), t.null]),
   sourceUrl: t.union([t.string, t.null]),
+  timestamp: t.union([t.number, t.null]),
```
Comment (lasryaric) [inline front/lib/upsert_document.ts:32]: Can we put `timestampMs`?

**Review by lasryaric — APPROVED** [review]
LGTM minus I would rename `timestamp` to `timestampMs` for clarity.

File: front/lib/upsert_document.ts (hunk 2 above)
Comment (spolu) [reply front/lib/upsert_document.ts:32]: We use timestamp everywhere for Data Source documents so I'd be in favor of valuing consistency over precision here?

**Pull request merged**